src/Backend/Core/Engine/Model.php

```diff
@@ -643,7 +643,7 @@ public static function getURLForBlock($module, $action = null, $language = null)
 
         // still no page id?
         if ($pageIdForURL === null) {
-            return self::getURL(404);
+            return self::getURL(404, $languag);
```

Comment:
but apart from that, isn't it a bit strange that we need to pass the language to show a 404 page ?